<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav26)"/>
  
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>
  
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg uav_name)_waypointflier_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  
  <group ns="$(arg uav_name)">
  
    <node pkg="nodelet" type="nodelet" name="kfupm_test" args="$(arg nodelet) kfupm_test/KfupmTest $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">
      
      <rosparam file="$(find kfupm_test_package)/config/kfupm_test.yaml" />
      
      <param name="uav_name" value="$(arg uav_name)" />
      <param name="odom_main_topic" value="/$(arg uav_name)/estimation_manager/odom_main" />
      <param name="control_service" value="/$(arg uav_name)/control_manager/goto" />
      <!-- <param name="control_service" value="/$(arg uav_name)/control_manager/goto_relative" /> -->
      
    </node>
    
  </group>
  
</launch>
